packages/language-support/src/textMateGrammar.ts

```diff
@@ -0,0 +1,103 @@
+import { keywordNames, operatorSymbols } from './lexerSymbols';
+
+const keywordRegex = new Array(...keywordNames.values()).join('|');
+const operatorsRegex = new Array(...operatorSymbols.values())
+  .map((v) => v.replace(/[.*+?^${}()|[\]\\/]/g, '\\$&'))
```

Comment:
Escapes reserved constructs in regexes, e.g. build the regex `\\*` instead of `*` which has the meaning of repeating the previous character indefinitely.